lib/characteristic.dart

```diff
@@ -20,7 +20,7 @@ class Characteristic extends InternalCharacteristic {
   /// The [Service] containing this characteristic.
   Service service;
```

Comment:
This should also be final, like the manager

Reply:
yes, indeed, in fact, a lot of vars and fields could be final, I only changed the ones I touched, so the PR remains relatively clean. But I generally agree with the "prefer final" idiom everywhere. I think it will be cleaner to do that refactor in another PR imho. :)